includes/Shopping/Shopify_Query.php

```diff
@@ -231,9 +231,35 @@ protected function fetch_remote_products( string $search_term ) {
 		 */
 		$result = json_decode( $body, true );
 
-		// TODO(#11268): Error handling.
 		if ( isset( $result['errors'] ) ) {
-			return new WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+			$wp_error = new WP_Error();
+
+			try {
+				$error      = array_shift( $result['errors'] );
+				$error_code = $error['extensions']['code'] ?? '';
```

Comment:
```suggestion
				foreach ( $result['errors'] as $error ) {
				    $error_code = $error['extensions']['code'] ?? '';
```

We could do this. WP_Error supports multiple error codes being added.

Reply:
Right -- 
Was just wondering your thoughts on that ...
https://github.com/GoogleForCreators/web-stories-wp/pull/11573#discussion_r883483688

Updated to use a loop now.